pkg/build/apis/build/validation/validation.go

```diff
@@ -68,6 +68,10 @@ func ValidateBuildUpdate(build *buildapi.Build, older *buildapi.Build) field.Err
 	if buildutil.IsBuildComplete(older) && older.Status.Phase != build.Status.Phase {
 		allErrs = append(allErrs, field.Invalid(field.NewPath("status", "phase"), build.Status.Phase, "phase cannot be updated from a terminal state"))
 	}
+
+	// lie about the old build's pushsecret value so we can allow it to be updated.
+	older.Spec.Output.PushSecret = build.Spec.Output.PushSecret
```

Comment:
Did you just mutate an incoming object which is sometimes backed by a cache inside of a validation function?  If you need to munge it, you should copy it.